### ISSUE TITLE
Add lgogdownloader version in .cache/lgogdownloader/gamedetails.json

### ISSUE THREAD

**Issue by a-detiste** [Sude-/lgogdownloader#66]
Hi,

I'm working on Debian's game-data-packager project that aims to automate commercial game assets installation.

https://github.com/a-detiste/game-data-packager

I just added extensive support for `lgogdownloader`.

To speed up operations, I read  `.cache/lgogdownloader/gamedetails.json`
directly. I don't know if you'll find this is an awesome or awfull idea. :sweat: 

Would it be possible to add lgogdownloader's version a the top level of the json data ?

This way old cache info could be discarded. This is maybe usefull for lgogdownloader itself.

![lgogdownloader](https://cloud.githubusercontent.com/assets/7994192/9930101/90841f24-5d31-11e5-8d4d-28bf70338095.png)

**Comment by ssokolow** [comment]
@a-detiste Do you have a page explaining your project in more detail? It looks like it might have some overlap with this experimental project I'm working on.

https://github.com/ssokolow/game_launcher

It's temporarily on hold because I'm just gearing up to do the final program project before my degree but, that aside, I'm always keeping my eyes open for opportunities to avoid unnecessarily duplicating effort. (Before I had to shift focus, I was working on adding unit tests so I could refactor the code into reusable modules and then, one-by-one, complete them enough to feel comfortable publishing them to PyPI)

**Comment by a-detiste** [comment]
Hi, here's the homepage:
https://wiki.debian.org/Games/GameDataPackager
and the list of games currently supported + TODO:
http://pkg-games.alioth.debian.org/game-data/

Each game has it's own ruleset in a yaml file:
http://anonscm.debian.org/cgit/pkg-games/game-data-packager.git/tree/data

This has been a Debian-only projects for years; but there isn't much difference between building a dumb .deb or dumb .rpm; so this work could be shared.

> https://github.com/ssokolow/game_launcher

Wow that's a nice PyGtk GUI. My plans currently look like an old xcdroast with a huge terminal widget.

**Comment by ssokolow** [comment]
@a-detiste  So more like my "just get something that works" frontend for lgogdownloader then? ;P

https://github.com/ssokolow/lgogd_uri

Seriously, though, it's really just a minimal test for the metadata extraction/inference code and smart icon scaling. (balancing pixel doubling and smooth scaling to minimize ugly blur when asked to do something like scaling a 16px icon up to 48px)

Any aesthetic appeal comes from the fact that I'm 50% UI/UX designer and don't want to deal with something ugly and uncomfortable while I prototype.

Once I've got the backends solid, I want to write a proper UI with Steam-like grouping support and some sort of as-you-type filtering (Possibly using the [frecency algorithm](https://developer.mozilla.org/en-US/docs/Mozilla/Tech/Places/Frecency_algorithm) from Firefox's address bar).

**Comment by a-detiste** [comment]
Goal here is to build a meta-Steam-like thing too, which give one's price quotes from GOG & Steam so users can pick what they prefer.

After games are installed, I'm used to just start them with KDE/LXQT omnibox (Alt-F2 + game name);
so each game should get it's .desktop file; so one can use it outside of the UI.

A beautiful launcher / browser would be nice for people who want to discover new games.

This should include distribution-provided games too.

I found this thing, but it's in FLTK; and can't even start/install the game, only browse:
https://screenshots.debian.net/package/goplay

> aesthetic

I've drawn some ugly & free icons :wink: being able to extract beautiful icons would better.
http://users.skynet.be/bs366950/icons/ ("own work")

As I do most of this work on a Raspberry Pi accessed from a WindowsXP pc through 
Shellinabox (the Rpi can power on the desktop PC when more power needed) , the desktop part will likely continue to progress at a snail pace.

**Comment by ssokolow** [comment]
@a-detiste In that case, I think you'd want to look into [Lutris](https://lutris.net/), which is also written in Python. (Which should make it clear why I'm focusing on the unique backend elements of mine at the moment.)

As far as price comparison, I generally rely on [IsThereAnyDeal.com](http://isthereanydeal.com/). It's a **very** big wheel to reinvent. (I even collaborated with the fellow who runs the site to produce a [userscript](https://github.com/ssokolow/itad_importer) to automate importing data from sites without APIs or public collection/wishlist views.)

**Comment by a-detiste** [comment]
>  IsThereAnyDeal.com

Wow that's huge, I didn't even know there were so many online shops...  I guess most doesn't sell for my country anyway.

> Lutris
>> "Like Steam or Desura, Lutris has two parts: a website and a client application, which communicate."

Yuck; what's the point of this, selling your play habits to some advertising agency ?

Where's the data ? I don't see any .yml file on the github repository. Is it like PlayOnLinux where game definitions are downloaded from somewhere ? I prefer having everything self-contained in a signed .deb that enable a real chain of trust.

> makeself_safeextract

I did the same thing ;-) in a lot more crude way

http://anonscm.debian.org/cgit/pkg-games/game-data-packager.git/tree/game_data_packager/make_template.py#n406



This is my first python project & I'm merely reproducing the same patterns as laid by original author; reading other projects will help me understand how a GUI is supposed to work. There is also the 'Ubuntu Software Center' that is a similar tool to check out.

**Comment by ssokolow** [comment]
> Yuck; what's the point of this, selling your play habits to some advertising agency ?
> 
> Where's the data ? I don't see any .yml file on the github repository. Is it like PlayOnLinux where game definitions are downloaded from somewhere ? I prefer having everything self-contained in a signed .deb that enable a real chain of trust.

It's similar to PlayDeb.net with the website serving as a catalogue of installable things.

The source to the website is here: http://www.playdeb.net/welcome/
The source to the client is here: https://github.com/lutris/lutris

As far as I can tell, the YAML is in the website's SQL database. I'm not sure about batch downloads, but you can download the YAML script for any individual game by pulling up the associated [catalogue page](https://lutris.net/games/0-ad/) on the website, clicking the drop-down beside the Install button, and choosing Download install script.

(Though, at the moment, most of the YAML that's offered either operates by integrating with 3rd-party mechanisms like the Steam client or simply integrates what was already provided by APT.)

I was more talking about taking advantage of the fact that the client's got a nice-looking UI, is written in Python, and is licensed under the GPLv3.

>  There is also the 'Ubuntu Software Center' that is a similar tool to check out.

Trust me. UI-wise, Ubuntu Software Center is far from the only tool of its kind. For example, there's the [GNOME Software Center](https://wiki.gnome.org/Apps/Software) PackageKit frontend, Antergos Linux is [working on one](https://www.phoronix.com/scan.php?page=news_item&px=Antergos-Poodle-Package-UI), and there are various other ideas brewing in the camp working on the [xdg-app](https://wiki.gnome.org/Projects/SandboxedApps) mechanism for Android-style sandboxed apps.

**Comment by a-detiste** [comment]
Yup, the YAML there are pretty bare. For game-data-packager we want to support all the original media or loose file too without relying on external tool. Each file get checksummed to make sure it will work with matching engine. This way we already found that someone's CD was scratched. (2 corrupted blocks in one file)  
We can also reject cracked games that scummvm will reject too.

The GUI is nice though !

I never used Ubuntu, I just learned the hard way I should make pretend when I do job interviews;
I do use some leaf packages like the one provided by PlayDeb.
